packages/aws-cdk-lib/aws-apigateway/test/deployment.test.ts

```diff
@@ -126,6 +126,21 @@ describe('deployment', () => {
     });
   });
 
+  test('"stage" can be set on the deployment', () => {
+    // GIVEN
+    const stack = new Stack();
+    const api = new apigateway.RestApi(stack, 'api', { deploy: false, cloudWatchRole: false });
+    api.root.addMethod('GET');
+
+    // WHEN
+    new apigateway.Deployment(stack, 'deployment', { api, stageName: 'dev' });
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::ApiGateway::Deployment', {
+      StageName: 'dev',
```

Comment:
Can we also have a test if the stage name is not specified in the Deployment, are we expecting default stage name `prod` to be used for deployment?

Reply:
If a new `Deployment` is created on its own, we do not expect `prod` to be used by default since the Deployment is not being created by the `RestApi`. 

It also seems that when `RestApi` automatically creates a new deployment, it doesn't set the `StageName` property (see [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-apigateway/lib/restapi.ts#L622-L625)). Instead it specifies the associated `deployment` when creating the stage (see [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-apigateway/lib/restapi.ts#L633). I think this makes sense because a stage can only be associated with one deployment, whereas a deployment could be deployed to multiple stages.